src/server/actions/coder.actions.ts

```diff
@@ -0,0 +1,35 @@
+'use server'
+
+import { getInfoForStudyId } from '@/server/db/queries'
+import { createUserAndWorkspace, getCoderWorkspaceUrl } from '../coder'
+import { Action, z } from './action'
+
+export const createUserAndWorkspaceAction = new Action('createUserAndWorkspaceAction', {})
+    .params(
+        z.object({
+            studyId: z.string().nonempty(),
+        }),
+    )
+    .middleware(async ({ params: { studyId } }) => await getInfoForStudyId(studyId))
+    .requireAbilityTo('create', 'Study')
+    .requireAbilityTo('load', 'IDE')
```

Comment:
don't need to check both, this can be only:

```suggestion
    .requireAbilityTo('load', 'IDE')
```
that's assuming the middleware call to `getInfoForStudyId` returns an object with a `researcherId` (I _think_ it does)